pkg/adapter/mtping/runner.go

```diff
@@ -127,7 +127,14 @@ func (a *cronJobsRunner) AddSchedule(source *sourcesv1.PingSource) (*time.Timer,
 		// The Date time is set before time.Now
 		duration := time.Until(date)
 
+		// If the scheduled time is earlier than the current time, fire events at least once.
+		if duration < 0 {
+			duration = 0 * time.Second
+		}
+
 		a.timer = time.AfterFunc(duration, a.cronTick(ctx, event))
+		source.Status.MarkCompleted()
```

Comment:
We need to mark it completed after cronTick  is actually done.

I would do something like this because I don't think this status update is actually persisted:

```golang
a.timer = time.AfterFunc(duration, a.markCompletedOnCompletion(source, a.cronTick(ctx, event)))

// ...

func (a *cronJobsRunner) markCompletedOnCompletion(before *sourcesv1.PingSource, f func()) func() {
  return func() {
      f() // dispatch event
      
     // pseudo code
     after := source.DeepCopy()
     after.Status.MarkCompleted()
     patch, err := duck.CreateMergePatch(before, after)
     // check err
     // patch ping source
  }
}
```

Reply:
Thanks for your advice!